crates/bevy_animation/src/lib.rs

```diff
@@ -639,106 +699,101 @@ fn apply_animation(
-            let curves = animation_clip.get_curves(*bone_id).unwrap();
-            let Some(target) = entity_from_path(root, path, children, names, cached_path) else {
-                continue;
-            };
-            any_path_found = true;
-            // SAFETY: The verify_no_ancestor_player check above ensures that two animation players cannot alias
-            // any of their descendant Transforms.
-            //
-            // The system scheduler prevents any other system from mutating Transforms at the same time,
-            // so the only way this fetch can alias is if two AnimationPlayers are targeting the same bone.
-            // This can only happen if there are two or more AnimationPlayers are ancestors to the same
-            // entities. By verifying that there is no other AnimationPlayer in the ancestors of a
-            // running AnimationPlayer before animating any entity, this fetch cannot alias.
-            //
-            // This means only the AnimationPlayers closest to the root of the hierarchy will be able
-            // to run their animation. Any players in the children or descendants will log a warning
-            // and do nothing.
-            let Ok(mut transform) = (unsafe { transforms.get_unchecked(target) }) else {
-                continue;
-            };
-            // SAFETY: As above, there can't be other AnimationPlayers with this target so this fetch can't alias
-            let mut morphs = unsafe { morphs.get_unchecked(target) }.ok();
-            for curve in curves {
-                // Some curves have only one keyframe used to set a transform
-                if curve.keyframe_timestamps.len() == 1 {
-                    match &curve.keyframes {
-                        Keyframes::Rotation(keyframes) => {
-                            transform.rotation = transform.rotation.slerp(keyframes[0], weight);
-                        }
-                        Keyframes::Translation(keyframes) => {
-                            transform.translation =
-                                transform.translation.lerp(keyframes[0], weight);
-                        }
-                        Keyframes::Scale(keyframes) => {
-                            transform.scale = transform.scale.lerp(keyframes[0], weight);
-                        }
-                        Keyframes::Weights(keyframes) => {
-                            if let Some(morphs) = &mut morphs {
-                                let target_count = morphs.weights().len();
-                                lerp_morph_weights(
-                                    morphs.weights_mut(),
-                                    get_keyframe(target_count, keyframes, 0).iter().copied(),
-                                    weight,
-                                );
-                            }
+    if animation.path_cache.len() != animation_clip.paths.len() {
+        animation.path_cache = vec![Vec::new(); animation_clip.paths.len()];
```

Comment:
do we need to make a whole new vec or can we just add/remove elements until the len is the same?

Reply:
It's possible to do a `clear()` followed by an `extend(…)`. The `Vec::new` here doesn't cause an allocation until something is pushed to it.